index.js

```diff
@@ -0,0 +1,161 @@
+    type: "movie",
+    boxOffice: "$460,998,507",
+    production: "Lucasfilm Ltd.",
+  },
+  {
+    id: 4,
+    title: "Harry Potter and the Half-Blood Prince",
+    year: 2009,
+    released: "15 Jul 2009",
+    runtime: "153 min",
+    genre: ["Action", "Adventure", "Family"],
+    director: "David Yates",
+    writer: "Steve Kloves, J.K. Rowling",
+    actors: ["Daniel Radcliffe", "Emma Watson", "Rupert Grint"],
+    plot: "As Harry Potter begins his sixth year at Hogwarts, he discovers an old book marked as 'the property of the Half-Blood Prince' and begins to learn more about Lord Voldemort's dark past.",
+    country: "United Kingdom",
+    poster:
+      "https://m.media-amazon.com/images/M/MV5BNzU3NDg4NTAyNV5BMl5BanBnXkFtZTcwOTg2ODg1Mg@@._V1_SX300.jpg",
+    imdbRating: 7.6,
+    imdbVotes: 492245,
+    boxOffice: "$302,305,431",
+    production: "Heyday Films, Warner Bros.",
+  },
+  {
+    title: "Harry Potter and the Sorcerer's Stone",
+    year: 2001,
+    released: "16 Nov 2001",
+    runtime: "152 min",
+    genre: ["Adventure", "Family", "Fantasy"],
+    director: "Chris Columbus",
+    writer: "J.K. Rowling, Steve Kloves",
+    actors: ["Daniel Radcliffe", "Rupert Grint", "Richard Harris"],
+    plot: "An orphaned boy enrolls in a school of wizardry, where he learns the truth about himself, his family and the terrible evil that haunts the magical world.",
+    country: "United Kingdom, United States",
+    poster:
+      "https://m.media-amazon.com/images/M/MV5BNjQ3NWNlNmQtMTE5ZS00MDdmLTlkZjUtZTBlM2UxMGFiMTU3XkEyXkFqcGdeQXVyNjUwNzk3NDc@._V1_SX300.jpg",
+    imdbRating: 7.6,
+    imdbVotes: 684604,
+    boxOffice: "$318,087,620",
+    production: "1492 Pictures, Heyday Films, Warner Brothers",
+  },
+];
+
+//1
+const arr = users.reduce((arr, { genre }) => {
+  return Array.from(new Set([...arr, ...genre]));
+}, []);
+
+//2
+const actor = users.reduce((arr, { actors }) => {
+  return Array.from(new Set([...arr, ...actors]));
+}, []);
+
+//3
+const reting = users.sort((a, b) => b.imdbRating - a.imdbRating);
+
+//4
+let objField = users.reduce((obj, { id, title, released, plot }) => {
+  return [...obj, { id, title, released, plot }];
+}, []);
+
+//6 С построчным фильтром
+let arrAutors = [...new Set(users.map((u) => u.writer))];
+//or С фильтром по одному автору
+let arrAutors2 = users.reduce((unique, { writer }) => {
+  unique = unique.includes(writer)
+    ? unique
+    : [...unique, writer].toString().split(",");
+  return [...new Set(unique)].map((el) => el.replace(/^ +/gm, ""));
+}, []);
+
+//7
+const fFilterFormTitle = (arr, string) => {
+  return arr.reduce((acc, film) => {
+    let { title } = film;
+    return title.includes(string) ? [...acc, film] : acc;
+  }, []);
+};
+
+//8
+const fFilterFromYear = (arr, Entryyear) => {
+  return arr.reduce((acc, film) => {
+    let { year } = film;
+    return year === Entryyear ? [...acc, film] : acc;
+  }, []);
+};
+
+//9
+const fFilterFromString = (arr, string) => {
```

Comment:
this task does not work correctly
'Black Widow' -> positive result,
but 'black widow' -> negative